aten/src/ATen/core/jit_type.h

```diff
@@ -1627,6 +1627,16 @@ struct CAFFE2_API ClassType : public NamedType {
   // These variants are not registered in the global class table.
   ClassTypePtr refine(at::ArrayRef<TypePtr> refined_slots) const;
 
+  TypePtr createWithContained(std::vector<TypePtr> contained_types) const override {
+    auto ptr = ClassType::create(name_, compilation_unit_);
```

Comment:
@driazati: Can you add a comment explaining the difference between between this method and `refine`? They seem identical accept with the subtyping test swapped, which is a pretty subtle distinction.